addons/website/models/ir_module_module.py

```diff
@@ -549,3 +550,184 @@ def _load_module_terms(self, modules, langs, overwrite=False):
+                if isinstance(value, list):
+                    items.extend(value)
+                else:
+                    items.extend(get_distinct_snippet_names(value))
+            return set(items)
+
+        create_count = 0
+        manifest = get_manifest(self.name)
+        # Configurator
+        configurator_snippets = manifest['snippet_lists']
+
+        # Generate general configurator snippet templates
+
+        create_values = []
+        # Every distinct snippet name across all configurator pages.
+        for snippet_name in get_distinct_snippet_names(configurator_snippets):
+            create_values.append(get_create_vals(
+                f"Snippet {snippet_name} for pages generated by the configurator",
+                snippet_name, '%s', 'configurator_%s'
+            ))
+        create_count += create_missing(create_values)
+
+        # Generate configurator snippet templates for specific pages
+
+        create_values = []
+        for page_name in configurator_snippets:
+            for snippet_name in configurator_snippets[page_name]:
+                create_values.append(get_create_vals(
+                    f"Snippet {snippet_name} for {page_name} pages generated by the configurator",
+                    snippet_name, 'configurator_%s', f'configurator_{page_name}_%s'
+                ))
+        create_count += create_missing(create_values)
+
+        # New page templates
+        templates = manifest['new_page_templates']
+
+        # Generate general new page snippet templates
+
+        create_values = []
+        # Every distinct snippet name across all new page templates.
+        for snippet_name in get_distinct_snippet_names(templates):
+            create_values.append(get_create_vals(
+                f"Snippet {snippet_name} for new page templates",
+                snippet_name, '%s', 'new_page_template_%s'
+            ))
+        create_count += create_missing(create_values)
+
+        # Generate new page snippet templates for new page template groups
+
+        create_values = []
+        for group in templates:
+            # Every distinct snippet name across all new page templates of group.
+            for snippet_name in get_distinct_snippet_names(templates[group]):
+                create_values.append(get_create_vals(
+                    f"Snippet {snippet_name} for new page {group} templates",
+                    snippet_name, 'new_page_template_%s', f'new_page_template_{group}_%s'
+                ))
+        create_count += create_missing(create_values)
+
+        # Generate new page snippet templates for specific new page templates within groups
+
+        create_values = []
+        for group in templates:
+            for template_name in templates[group]:
+                for snippet_name in templates[group][template_name]:
+                    create_values.append(get_create_vals(
+                        f"Snippet {snippet_name} for new page {group} template {template_name}",
```

Comment:
Followup of https://github.com/odoo/odoo/pull/126719#discussion_r1321604795
Can we go with `%r` everywhere we log something or create a name which uses a word linked to a "ressource" in the middle of a sentence?
For instance:
![image](https://github.com/odoo/odoo/assets/30048408/5ea696a3-9674-4fb7-b420-206cc15639f4)
That's the name of a view, it's really hard to figure that `about` and `full` are the meaningful parts here: it's the `full` template of the `about` section.
This would become 
![image](https://github.com/odoo/odoo/assets/30048408/815f7639-cecd-45a2-85b4-c923ac81a369)